pkg/maps/nat/stats/cell.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package stats
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cilium/cilium/pkg/maps/nat"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/index"
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/pflag"
+)
+
+var Cell = cell.Module(
+	"nat-stats",
+	"Aggregates stats for NAT maps",
+	metrics.Metric(newMetrics),
+	cell.ProvidePrivate(newTables),
+	cell.Provide(
+		func(m Metrics) natMetrics {
+			return m
+		},
+		newStats,
+	),
+	cell.Config(Config{
+		NATMapStatInterval:       30 * time.Second,
+		NatMapStatKStoredEntries: 32,
```

Comment:
How did you decide on these defaults?

Reply:
For NatMapStatKStoredEntries: this is just how many of the top-k entries we're planning on storing in statedb (and eventually an optional metric) by default. 32 seemed reasonable.

NATMapStatInterval was also just based on a reasonable interval, however I should expand on this and do some benchmarking based on this interval.

Reply:
It would be good to leave either a code comment or reference it in the commit msg just to have a record.

Reply:
To follow up here, here's some basic testing that these settings are and the overall change set don't cause any apparant memory regressions in a "general" case (in this test case I ran EGW while forwarding 40k connections with my changes and without).

![control](https://github.com/cilium/cilium/assets/5881740/daafe818-d02e-4071-8a3a-a41ced9de1a6)
![test](https://github.com/cilium/cilium/assets/5881740/7dc69f66-7b2e-4bd8-888e-3fa82d333261)

Reply:
I'll add some comments explaining this context